pkg/hostagent/pods.go

```diff
@@ -386,3 +387,19 @@ func (agent *HostAgent) podDeletedLocked(obj interface{}) {
 	}
 	agent.epDeleted(&u)
 }
+
+func (agent *HostAgent) cniEpDelete(cniKey string) {
+	agent.indexMutex.Lock()
+	defer agent.indexMutex.Unlock()
+	epUuid, ok := agent.cniToPodID[cniKey]
+	if !ok {
+		agent.log.Warnf("cniEpDelete: PodID not found for %s", cniKey)
+		return
+	}
+	delete(agent.cniToPodID, cniKey)
+
+	if _, ok := agent.opflexEps[epUuid]; ok {
+		delete(agent.opflexEps, epUuid)
+		agent.scheduleSyncEps()
```

Comment:
one more thing that looks like a design problem is why do we sync / walk all the eps every time on add or delete. The sync should happen either on a timer or when agent restarts. when we add or delete we exactly know which pod it is. Not sure why we need to walk that list.

Reply:
Yes, this is per current design. The alternative is to generate deltas and send it to the sync thread. That can have races which will need to be explicitly addressed. That's probably why the original implementer chose the current approach.